### ISSUE TITLE
js to js waku relay publish does not seem reproducable in test

### ISSUE THREAD

**Issue by D4nte** [status-im/js-waku#7]
## Problem

Not able to confirm that publication of messages using waku relay works between two nodes using js-waku despite it being working between nim-waky and js-waku nodes.

## Solution

Further investigate the issue

## Acceptance criteria

This test passes https://github.com/status-im/js-waku/blob/46869e998d710d410450fff66e0ba83ca99b1d4c/src/lib/waku_relay.spec.ts#L75

**Comment by oskarth** [comment]
Any idea what could be causing this? Does stock JS gossipsub example work JS-JS?

**Comment by D4nte** [comment]
> Any idea what could be causing this? Does stock JS gossipsub example work JS-JS?

I have found the issue now, this is what I mentioned today in the chat. Currently writing code to fix it.

gossipsub check against the gossipsub protocol in various places (when joining, when publishing, at heartbeat, etc).

I did override to use waku protocol in some places, but not all. So in some cases it was working, but in the case of js<>js tests, both node subscribe at the same to the topic so it goes through a different flow than all nim<>js tests. In this flow, the fact that I did not replace the gossipsub protocol with waku relay in `_publish` created the issue.
